tests/ztest/src/ztest.c

```diff
@@ -23,6 +26,15 @@ static int cleanup_test(struct unit_test *test)
 	int mock_status;
 
 	mock_status = _cleanup_mock();
+
+	#ifdef KERNEL
+	/* we need to remove the ztest_thread information from the timeout_q.
+	 * Because we reuse the same k_thread structure this would
+	 * causes some problems.
+	 */
```

Comment:
may be you have change your comment. I don't see a you are directly touching timeout queue and even your commit message as well.